lib/simple_segment/operations/operation.rb

```diff
@@ -40,6 +40,14 @@ def check_identity!
           raise ArgumentError, 'user_id or anonymous_id must be present'
         end
       end
+
+      def timestamp(timestamp)
+        if timestamp.respond_to?(:iso8601)
+          timestamp.iso8601
+        else
+          Time.iso8601(timestamp)
```

Comment:
@mickey Hey, let me know if you're still interested in this. Otherwise I can cherry-pick your commit to a new branch.